packages/unzip.rb

```diff
@@ -39,11 +28,9 @@ def self.patch
   end
 
   def self.build
-    if ARCH == "armv7l"
-      system "make -f unix/Makefile linux_noasm"
-    else
-      system "make -f unix/Makefile generic"
-    end
+    # LCHMOD is not supported in linux.
+    system 'echo "CFLAGS=-DNO_LCHMOD" >> flags'
```

Comment:
Would this work also?
```
system "CFLAGS=-DNO_LCHMOD make -f unix/Makefile generic"
```

Reply:
That doesn't work. :( The unzip Makefile invokes the CFLAGS from a flags file. It's weird and utterly nonstandard.